src/pool/srv_pool.c

```diff
@@ -5022,31 +5121,59 @@ ds_pool_svc_delete_acl(uuid_t pool_uuid, d_rank_list_t *ranks,
 	return rc;
 }
 
-static void
-pool_svc_reconfigure_replicas(struct pool_svc *svc, int svc_rf, struct pool_map *map)
+static bool
+reconf_canceled(struct pool_svc_reconf *reconf)
 {
-	d_rank_list_t	*current;
-	d_rank_list_t	*to_add;
-	d_rank_list_t	*to_remove;
-	d_rank_list_t	*new;
-	int              rc;
+	bool canceled;
+
+	ABT_mutex_lock(reconf->psc_mutex);
+	canceled = reconf->psc_canceled;
+	ABT_mutex_unlock(reconf->psc_mutex);
+	return canceled;
+}
+
+static void
+pool_svc_reconf_ult(void *arg)
+{
+	struct pool_svc		*svc = arg;
+	struct pool_svc_reconf	*reconf = &svc->ps_reconf;
+	d_rank_list_t		*current;
+	d_rank_list_t		*to_add;
+	d_rank_list_t		*to_remove;
+	d_rank_list_t		*new;
+	int			 rc;
+
+	D_DEBUG(DB_MD, DF_UUID": begin\n", DP_UUID(svc->ps_uuid));
+
+	/* When there are pending events, the pool map may be unstable. */
+	while (events_pending(svc)) {
+		dss_sleep(3000 /* ms */);
+	       	if (reconf_canceled(reconf)) {
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs